sql/core/src/test/scala/org/apache/spark/sql/ShowCreateTableSuite.scala

```diff
@@ -176,19 +177,32 @@ abstract class ShowCreateTableSuite extends QueryTest with SQLTestUtils {
       val createTable = "CREATE TABLE `t1` (`a` STRUCT<`b`: STRING>)"
       sql(s"$createTable USING json")
       val shownDDL = getShowDDL("SHOW CREATE TABLE t1")
-      assert(shownDDL == "CREATE TABLE `default`.`t1` (`a` STRUCT<`b`: STRING>)")
+      assert(shownDDL.mkString(" ") ==
+        "CREATE TABLE `default`.`t1` ( `a` STRUCT<`b`: STRING>) USING json")
 
       checkCreateTable("t1")
     }
   }
 
-  protected def getShowDDL(showCreateTableSql: String): String = {
-    val result = sql(showCreateTableSql)
-      .head()
-      .getString(0)
-      .split("\n")
-      .map(_.trim)
-    if (result.length > 1) result(0) + result(1) else result.head
+  test("SPARK-36012: Add NULL flag when SHOW CREATE TABLE") {
+    val t = "SPARK_36012"
+    withTable(t) {
+      sql(
+        s"""
+           |CREATE TABLE $t (
+           |  a bigint NOT NULL,
+           |  b bigint
+           |)
+           |USING ${classOf[SimpleInsertSource].getName}
+        """.stripMargin)
+      val showDDL = getShowDDL(s"SHOW CREATE TABLE $t")
+      assert(showDDL.mkString(" ") == s"CREATE TABLE `default`.`$t` ( `a` BIGINT NOT NULL," +
+        s" `b` BIGINT) USING ${classOf[SimpleInsertSource].getName}")
+    }
+  }
+
+  protected def getShowDDL(showCreateTableSql: String): Array[String] = {
+    sql(showCreateTableSql).head().getString(0).split("\n").map(_.trim)
```

Comment:
why can't we always call `.mkString(" ")` here?

Reply:
I think @Peng-Lei  doesn't want to include  the `USING text TBLPROPERTIES ( 'transient_lastDdlTime...` in the test in `HiveShowCreateTableSuite`. I guess maybe always call `.mkString(" ")` here and trim off the `USING text ...` using something like `shownDDL.substring(0, shownDDL.indexOf(" USING")` in the Hive test?

Reply:
Yeah. In `HiveShowCreateTableSuite` it compare the `Array(0) + Array(1)`. I do not want to include the `USING text TBLPROPERTIES`, because the test fail. I just think `getShowDDL` return Array[String] is flexible, because if we want to use `Array` to assert, it just ok. If we want to use `String` to assert, just `mkString()`. @cloud-fan @huaxingao